x-pack/test/observability_functional/apps/observability/alerts/workflow_status.ts

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import expect from '@kbn/expect';
+import { FtrProviderContext } from '../../../ftr_provider_context';
+
+export default ({ getService }: FtrProviderContext) => {
+  const esArchiver = getService('esArchiver');
+
+  describe('alert workflow status', function () {
+    this.tags('includeFirefox');
+
+    const observability = getService('observability');
+    const retry = getService('retry');
+
+    before(async () => {
+      await esArchiver.load('x-pack/test/functional/es_archives/observability/alerts');
+      await observability.alerts.navigateToTimeWithData();
+    });
+
+    after(async () => {
+      await esArchiver.unload('x-pack/test/functional/es_archives/observability/alerts');
+    });
+
+    it('is filtered to only show "open" alerts by default', async () => {
```

Comment:
Just a note:
This test asserts something that is the outcome of the workflow status filter bar state. 
While the test right below tests that an individual alert in the table can have its state changed. 

For me those actions are orthogonal to each other. Another way to group this is for example like this:
Test default workflow status alert count
Test acknowledged workflow status alert count
Test recovered workflow status alert count

Test table action mark open
Test table action mark acknowledged
Test table action mark recovered

But if I understood your previous comment, you read this test file as a set of tests that are "chronological" in how a user would go through the page? (first see the landing state, mark one alert as acknowledged, then narrow the table to only show acknowledged). 
For me, since they are different test cases they don't have any chronology between them (some principle about that tests should not rely on each other) and as such it would make more sense to group them by action taken (filter vs change state). If we want communicate a chronology then I would put more actions into a single test case.

Does that make sense?

(This is just a lot of opinions and me trying to build an understanding of how things are done!)

Reply:
Nit: This test name is still a bit unclear with regards to **what** "is filtered". Maybe the word table is needed here as well?
"the table is filtered to only show..."?

Since `it`refers to 'alert workflow status' I think it makes sense to define that a bit more. 

Usage of describe is another topic for later but do we aim to follow BDD patterns? Or would we prefer to use `test` instead of `it`?

I don't think this needs to be changed, more looking to discuss patterns.

Reply:
> But if I understood your previous comment, you read this test file as a set of tests that are "chronological" in how a user would go through the page? (first see the landing state, mark one alert as acknowledged, then narrow the table to only show acknowledged).
For me, since they are different test cases they don't have any chronology between them (some principle about that tests should not rely on each other) and as such it would make more sense to group them by action taken (filter vs change state). If we want communicate a chronology then I would put more actions into a single test case.
> 
> Does that make sense?

It does, but here we're working within the environment that the Kibana test tooling and conventions create. Building the tests like this is in line with the [documented approach](https://www.elastic.co/guide/en/kibana/current/development-tests.html).

I can see arguments for both the "isolated test case" and the "sequential workflow" approach. We're used to the former from unit tests. The latter offers some advantages as well, though. It allows us to take a user-centric testing approach which can cover long workflows without sacrificing test result reporting precision. By treating each suite as one test and each test case as a step your test result naturally records the step at which the workflow has failed.

Reply:
> I don't think this needs to be changed, more looking to discuss patterns.

I like discussing patterns too, but in this case I suggest we separate the discussion of patterns from the effort of ramping up our test coverage as quickly as possible.

Reply:
Thanks for the added info, makes sense. I imagine it also plays into how we load data per suite (as in the data needed for this workflow). It's helpful to know that I should read these tests like "user does `it` then `it`"!

Let's take further pattern discussions in another channel :)